src/main/java/org/jabref/logic/bibtex/comparator/BibDatabaseDiff.java

```diff
@@ -89,6 +89,10 @@ private static List<BibEntryDiff> compareEntries(List<BibEntry> originalEntries,
                 }
             }
 
+            // Prevent IndexOutOfBoundException
+            if (newEntries.isEmpty()) {
+                return differences;
+            }
```

Comment:
Shouldn't that go up to the beginning of the method? differences is not filled up to now, newEntries is also not modified.